### ISSUE TITLE
No Maven output color in netbeans consoles

### ISSUE THREAD

**Issue by albilu** [apache/netbeans#6210]
### Apache NetBeans version

Apache NetBeans 18

### What happened

Unlike all other main IDES Netbeans is unable to print Maven logs with color.

Netbeans:
![image](https://github.com/apache/netbeans/assets/35330562/d8432dcd-c04b-4a07-b1a4-d3ea65dc4c7d)

IntelliJ IDEA:
![image](https://github.com/apache/netbeans/assets/35330562/9a4f0100-a732-4bda-825f-e5020c3b2977)

### How to reproduce

Run a Maven project in Netbeans. 

### Did this work correctly in an earlier version?

No / Don't know

### Operating System

Windows/Linux

### JDK

Any

### Apache NetBeans packaging

Apache NetBeans provided installer

### Anything else

Related issues:
https://github.com/fusesource/jansi/issues/87
https://peter.quantr.hk/2018/12/solved-the-root-cause-why-netbeans-cant-print-colored-text-using-latest-maven/#page-content

### Are you willing to submit a pull request?

No

**Comment by neilcsmith-net** [comment]
The workaround is given in the Jansi issue you linked, and also the trade off that introduces (eg. other aspects of output parsing no longer work).

Moving this to a feature request.

**Comment by mbien** [comment]
I am not planning to implement this, adding https://github.com/apache/netbeans/labels/Contribution%20welcome label.

**Comment by asbachb** [comment]
I wonder if it would be useful to have an alternative to the current output panel. For `mvnd` the output is not optimal as well. When you launch it from the embedded terminal it looks quite good but lacks features like folding (which I actually never used).
Maybe we could introduce an option to execute maven on the terminal?

**Comment by Chris2011** [comment]
Then we can also have an alternative output panel for git, ant, npm, etc... so I would like to let them as it is, because we already have more than just maven.

**Comment by neilcsmith-net** [comment]
@Chris2011 well, we could look at an option to run any external process in the terminal rather than the output window - ie. ant and npm, but not git (while using jgit).

**Comment by neilcsmith-net** [comment]
@Chris2011 of course, and for macOS on Arm for that matter. Having a (non-default) option isn't that much of a rabbithole, if someone wanted to work on it.

**Comment by asbachb** [comment]
I wonder if it's an option to ship with a java terminal emulator like https://github.com/JetBrains/jediterm

**Comment by Chris2011** [comment]
> I wonder if it's an option to ship with a java terminal emulator like https://github.com/JetBrains/jediterm

Ui that looks interesting :D. Maybe will have a look later. Another project......

**Comment by moquinr** [comment]
> @neilcsmith-net but then we need real implementation of native terminals also for Windows users w/o using cygwin or anything else. And this is another rabbithole. IMHO.

Why can't it use a wsl 2 terminal if it's enabled, like Docker Desktop does... then you have a full linux shell available... not sure if the new Windows Terminal support is a possibility...

**Comment by Chris2011** [comment]
It is somehow hardcoded that if windows it searches for pty smth like that and if this is not installed like cygwin, you got an info. WSL is not working like pty it is another implementation. I dunno how hard it will be for WSL but if we can reuse some other tools like pty4j or jediterm or whatever, that could be good.

**Comment by Chris2011** [comment]
We don't need to reinvent the wheel again, luckily. But someone needs to start on it.

**Comment by rrangelo** [comment]
Hi all!

I recommend use: [JLine](https://github.com/jline/jline3). It's FOSS and is used by Apache Maven.

I was looking info about Pty4J and I asked to AI create a comparison table. I share them. I hope it can be help you.

![image](https://github.com/apache/netbeans/assets/68767628/531b4732-c199-4876-99c9-d05100a34de7)
![image](https://github.com/apache/netbeans/assets/68767628/fb505969-adfb-4802-a12a-9c0cd4696878)
![image](https://github.com/apache/netbeans/assets/68767628/94ca8fc9-0966-4130-9951-a825160f6583)
